src/classes/WalletConnectors/WalletConnector.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * An abstract class that provides the interface to implement a Wallet Connector.
+ */
+export abstract class WalletConnector<Provider = any> {
+    /** Unique wallet connector id */
+    public abstract readonly id: string
+    /** Wallet connector name */
+    public abstract readonly name: string
+
```

Comment:
what do you think about adding an image property?

Reply:
Actually not a bad idea :+1: I'll make it optional btw...